### ISSUE TITLE
Standardize UI and mechanism for help icons

### ISSUE THREAD

**Issue by jonatanschroeder** [PrairieLearn/PrairieLearn#7769]
As discussed in https://github.com/PrairieLearn/PrairieLearn/pull/6271#discussion_r1196959374 and on Slack, the interface for help buttons is not standard across PL/PT. In particular, the icons are not consistent, the use of tooltips/popovers/modals is not consistent, accessibility is not always considered.

**Comment by ThePageGuy** [comment]
Hello, just checking this GitHub Issue and wanting to confirm some things. If understood correctly the approach this GitHub Issue is divided into three part:
- One is to make sure all the help buttons use the same “?” Icon, which was mentioned on this comment https://github.com/PrairieLearn/PrairieLearn/pull/6271#discussion_r1197244319
- Second it looks like there is not a consistency whether for the help button to use tooltips, popovers, or modals as to make this help buttons work. We can select one base on the last part.
- Third is regarding the help button accessibility, and this would change depending what option we choose from part two:
    - Tooltip is the one that may have more problem with accessibility since it opens when hovered, and base on what the WCAG 1.4.13 states: All content made visible should be persistent, meaning it should not disappear without user action. Tooltip Doc: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/tooltip_role
    - Popover is the one, in my opinion, that have the best value in terms of accessibility, since when a popover is shown, the keyboard focus navigation order is updated so that the popover is next in the sequence. It also allow screen readers to make sense of the relationship between the invoker and the popover, by using some implicit tag. Popover Docs: https://developer.mozilla.org/en-US/docs/Web/API/Popover_API/Using
    - Modal is another one good for accessibility, however in the case for a help button it may not be the best approach, since when interacted with the dialog tends to fill 100% of the screen. (Maybe there a tag that can change this, but haven’t found one yet) Modal Docs: https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/

**Comment by jonatanschroeder** [comment]
I'll let others comment on the direction to move on, but I want to add that another problem with modals is that you can't open modals on top of each other, and some help buttons are already inside modals. 

As for tooltips, I believe screen readers have means to announce them if the button gets focus. Also, bootstrap tooltips show up when the button acquires focus, regardless of mouse position. I may be wrong, please test this.

**Comment by nwalters512** [comment]
Yeah, unfortunately the original issue here is under-specified, so I'm really not sure what we were thinking when this was opened.

I agree that standardizing on the `?` icon is a good move, though I'm not actually sure where we deviate from that.

I think it's OK that we use any/all of tooltips, popovers, and modals, as they have different use cases:

- Tooltips are good for short bits of text, like descriptions on form inputs.
- Popovers are good for longer bits of text that require HTML formatting, like the popover that describes the scoring rules on instance questions.
- Modals are good for very long/complex information, like the help modal on the issues search page.

I'm not aware of any accessibility issues with any of our current implementations. It's possible that there were some back when this issue was first authored.

My advice: we should start by describing concretely how we think the current design is deficient. This means picking out specific examples of e.g. inconsistent button styles or iconography, places where we think the wrong UI component (tooltip/popover/modal) is used, or places where help information isn't accessible. Then we can talk specifically about what kind of changes we'd want to make.
